File: 프로그래머스/0/120866. 안전지대/안전지대.swift
import Foundation

func solution(_ board:[[Int]]) -> Int {
    let n = board.count
    var ans = 0
    let dr = [-1, 0, 1, -1, 0, 1, -1, 0, 1]
    let dc = [-1, -1, -1, 0, 0, 0, 1, 1, 1]
    
    func isBomb(_ row: Int, _ col: Int) -> Bool {
        if row < 0 || col < 0 || row >= n || col >= n { return false }
        return board[row][col] == 1
    }
    
    for i in 0..<n {
        for j in 0..<n {
            for k in 0..<9 {
                if isBomb(i + dr[k], j + dc[k]) { 
                    ans += 1
                    break
                }
            }
        }
    }
    
    return n * n - ans
}

File: 프로그래머스/0/120866. 안전지대/안전지대.js
function solution(board) {
    var b = board.map(v => [...v]);
    const n = board.length;
    const dx = [-1, 0, 1, -1, 0, 1, -1, 0, 1];
    const dy = [-1, -1, -1, 0, 0, 0, 1, 1, 1];
    
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (board[i][j] == 1) {
                for (let k = 0; k < 9; k++) {
                    var x = i + dx[k];
                    var y = j + dy[k];
                    if (x > -1 && y > -1 && x < n && y < n) {
                        b[x][y] = 1;
                    }
                }
            }
        }
    }
    return b
        .reduce((acc, now) => [...acc, ...now], [])
        .filter(v => v == 0)
        .length;
}